src/Core/Usecase/Message/ReceiveMessage.php

```diff
@@ -173,10 +173,13 @@ protected function createPost(Entity $message)
 		$content = $message->message;
 
 		if ($message->additional_data) {
-			if (isset($message->additional_data['form_id'])) {
-				$form_id = $message->additional_data['form_id'];
+			// @todo inject
+			$source = app('datasources')->getSource($message->data_source);
+			$formId = $source->getInboundFormId();
```

Comment:
@willdoran I've moved this out of the message data since its really tied to the data source. 
I think this makes sense?

Reply:
If it still works, sure.